modules/crypt.cpp

```diff
@@ -65,9 +199,17 @@ class CCryptMod : public CModule {
         AddCommand("ListKeys", static_cast<CModCommand::ModCmdFunc>(
                                    &CCryptMod::OnListKeysCommand),
                    "", "List all keys");
+        AddCommand("KeyX", static_cast<CModCommand::ModCmdFunc>(
+                                 &CCryptMod::OnKeyXCommand),
+                   "<Nick>", "Start a DH1080 key exchange with nick");
     }
 
-    ~CCryptMod() override {}
+    ~CCryptMod() override {
+        if (dh) {
+            DH_free(dh);
+            dh = NULL;
```

Comment:
I didn't actually test this, but I imagine something like this:
```c++
std::unique_ptr<DH, decltype(&DH_free)> m_dh;
```
This would then require the constructor of CCryptMod to do something like `m_dh(nullptr, DH_free)` to initalize this object. To construct a new DH instance, use `m_dh.reset(DH_new())`. In other places, it should just be possible to use `m_dh` as if it had type `DH*`.

@DarthGandalf Please correct me if I am wrong (I don't think I ever used this before).